pkg/transaction/payload.go

```diff
@@ -141,10 +143,15 @@ func (p *TokenTransferPayload) Deserialize(data []byte) (int, error) {
 }
 
 func (p *SmartContractPayload) Serialize() ([]byte, error) {
-	buf := make([]byte, 0, len(p.ContractName)+len(p.CodeBody)+6) // 1 + 1 + 4 bytes for headers
+	buf := make([]byte, 0, len(p.ContractName)+len(p.CodeBody)+7) // 1 + 1 + 1 + 4 bytes for headers
```

Comment:
Does this work on-chain?

Reply:
That's what VersionedSmartContract wire format looks like. Additional ClarityVersion following the PayloadType 6